main.go

```diff
@@ -73,13 +72,18 @@ func main() {
 			"Enabling this will ensure there is only one active controller manager.")
 	flag.IntVar(&concurrent, "concurrent", 4, "The number of concurrent kustomize reconciles.")
 	flag.DurationVar(&requeueDependency, "requeue-dependency", 30*time.Second, "The interval at which failing dependencies are reevaluated.")
-	flag.StringVar(&logLevel, "log-level", "info", "Set logging level. Can be debug, info or error.")
-	flag.BoolVar(&logJSON, "log-json", false, "Set logging to JSON format.")
 	flag.BoolVar(&watchAllNamespaces, "watch-all-namespaces", true,
 		"Watch for custom resources in all namespaces, if set to false it will only watch the runtime namespace.")
+	flag.Bool("log-json", false, "Set logging to JSON format.")
+	flag.CommandLine.MarkDeprecated("log-json", "Please use --log-level=json instead.")
```

Comment:
Where is the `--log-level` flag defined?

Reply:
It is a typo, it should be log format, these flags come from our pkg/runtime

cc @hiddeco this is what i meant that flags from packages are magically confusing :)

Reply:
I was suspecting that the flag was set in pkg. I would have to agree that it is confusing. Sure it is nice that it happens auto magically but its better at times to be explicit in what is happening.

Reply:
> Sure it is nice that it happens auto magically but its better at times to be explicit in what is happening.

Not if all controllers are expected to follow the same contract, the magic that happens here assures they do and are always aligned. Which makes the maintainer burden of thinking about the magic more justified to me than having misaligned flags across controllers.

Reply:
As long as there is a good reason I think it is fine to decide that the pros outweigh the cons. My thought is just that the log options will probably not change often or ever, but I may be wrong. Now that I know that this happens it wont affect me.